test/server/config_validation/dispatcher_test.cc

```diff
@@ -63,14 +63,10 @@ TEST_P(ConfigValidation, CreateScaledTimer) {
 // Make sure that creating DnsResolver does not cause crash and each call to create
 // DNS resolver returns the same shared_ptr.
 TEST_F(ConfigValidation, SharedDnsResolver) {
-  std::vector<Network::Address::InstanceConstSharedPtr> resolvers;
-  auto dns_resolver_options = envoy::config::core::v3::DnsResolverOptions();
-
-  Network::DnsResolverSharedPtr dns1 =
-      dispatcher_->createDnsResolver(resolvers, dns_resolver_options);
+  auto typed_dns_resolver_config = envoy::config::core::v3::TypedExtensionConfig();
```

Comment:
If typed extension config is empty what resolver will it create?

Reply:
Good catch. That will cause exception be thrown. We need to create an empty cares typed config here.

Reply:
My initial change to address this comments always create an empty c-ares DNS resolver. This differs from the original behavior in Apple system, which creates AppleDnsResolver.  Commited a new change https://github.com/envoyproxy/envoy/pull/17479/commits/be9ea18f5e96deab9ebc7857d19787adff168787
In this change, a generic makeEmptyDnsResolver() function is called here also a few other similar places to create Empty DNS resolver based on build system and run-time configurations. Please take a look.